### ISSUE TITLE
FlatList when set `horizontal={true}` ,ItemSeparatorComponent can't be shown correctly

### ISSUE THREAD

**Issue by aiynmm** [facebook/react-native#16518]
#

### Is this a bug report?

Maybe yes

### Have you read the [Contributing Guidelines](https://facebook.github.io/react-native/docs/contributing.html)?

yes

### Environment

 Environment:
    OS:  window 10
    Node:  8.1.3
    Yarn:  0.27.5
    npm:  5.0.3
    Watchman:  NO
    Xcode: NO
    Android Studio:  NO

  Packages: (wanted => installed)
    react-native: 0.49.3 => 0.49.3
    react: 16.0.0-beat.5 => 16.0.0--beat.5

  Target Platform: Android

### Steps to Reproduce

(Write your steps here:)

1.Set `<FlatList horizontal={true}/>` 
2.Use `ItemSeparatorComponent={}` set separator

### Expected Behavior

I want to see the ItemSeparatorComponent is shown correctly!

### Actual Behavior

if ItemSeparatorComponent has no height ,it won't be shown,

![qq 20171024151021](https://user-images.githubusercontent.com/15321155/31930356-07d0aa72-b8d2-11e7-9ac6-34b8f9a46a34.png)

but when set height to ItemSeparatorComponent ,it will be shown below the data!

![qq 20171024151116](https://user-images.githubusercontent.com/15321155/31930361-0bb7adc0-b8d2-11e7-92c0-85d43e7cd9d9.png)

### Reproducible Demo

demo:
```
let data = [];
for (let i = 0; i < 100; i++) {
  data.push({key: i, num: i});
}
```

```
space(){
    return(<View style={{height: 50, width: 2, backgroundColor: 'black'}}/>)
}

render() {
    return (
      <View style={{height: 50, backgroundColor: 'lightgreen'}}>
        <FlatList
          data={data}
          ItemSeparatorComponent={this.space}
          horizontal={true}
          renderItem={({item}) =>
            <Text style={{fontSize: 20, textAlign: 'center'}}>{item.num}</Text>
          }
        />
      </View>
    )
  }
```

**Comment by dgruseck** [comment]
This issue should be fixed with v0.50, see #15865.
Can you try it with the release candidate?

**Comment by aiynmm** [comment]
@dgruseck OK，I’ll try it later!
